### PR TITLE
Changes for mobile apps (agent & cus-regs)

### PR DESCRIPTION
**Pull request by InensusDev**

### CODE REVIEW

File: Website/htdocs/mpmanager/app/Services/PersonService.php [hunk 1]
```diff
@@ -196,4 +197,10 @@ public function createFromRequest(Request $request): Model
 
         return $person;
     }
+
+    public function getByPhoneNumber($phoneNumber)
```
Comment (alchalade) [inline Website/htdocs/mpmanager/app/Services/PersonService.php:201]: @InensusDev ,
return type should be `?Person`

File: Website/htdocs/mpmanager/app/Traits/RestExceptionHandler.php [hunk 2]
```diff
@@ -27,9 +30,14 @@ trait RestExceptionHandler
      */
     protected function getJsonResponseForException(Request $request, Exception|Throwable $e)
     {
-
         $response = null;
```
Comment (alchalade) [inline Website/htdocs/mpmanager/app/Traits/RestExceptionHandler.php:33]: @InensusDev no need for this as **default** will cover all cases.

File: Website/htdocs/mpmanager/app/Utils/ApplianceInstallmentPayer.php [hunk 3]
```diff
@@ -38,21 +38,66 @@ public function initialize(TransactionDataContainer $transactionData): void
...
+    }
+
+    private function getInstallments($customer)
+    {
+        $loans = $this->appliancePersonService->getLoanIdsForCustomerId($customer->id);
+
+        return $this->applianceRateService->getByLoanIdsForDueDate($loans);
+    }
+
+    private function pay(mixed $installments, mixed $customer): void
```
Comment (alchalade) [inline Website/htdocs/mpmanager/app/Utils/ApplianceInstallmentPayer.php:98]: @InensusDev  isn't `$installments` array?

File: Website/htdocs/mpmanager/routes/resources/CustomerRegistrationApp.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    Route::get('/people', 'PersonController@index');
+    Route::get('/manufacturers', 'ManufacturerController@index');
+    Route::get('/meter-types', 'MeterTypeController@index');
+    Route::get('/tariffs', 'MeterTariffController@index');
+    Route::get('/cities', 'CityController@index');
+    Route::get('/connection-groups', 'ConnectionGroupController@index');
+    Route::get('/connection-types', 'ConnectionTypeController@index');
+    Route::get('/sub-connection-types', 'SubConnectionTypeController@index');
+    Route::post('/', static function (AndroidAppRequest $r) {
+        try {
```
Comment (alchalade) [inline Website/htdocs/mpmanager/routes/resources/CustomerRegistrationApp.php:29]: @InensusDev could you move that whole logic to MobileAppService without refactoring?

File: Website/htdocs/mpmanager/app/Utils/ApplianceInstallmentPayer.php (hunk 3 above)
Comment (InensusDev) [reply Website/htdocs/mpmanager/app/Utils/ApplianceInstallmentPayer.php:98]: its a collection

**Review by alchalade — APPROVED** [review]

**Pull request merged**